framework/base/Profiler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\base;
+
+use Yii;
+use yii\log\Logger;
+
+/**
+ * Profiler is the class for application profiling.
+ *
+ * Profiler is used as a core component of Yii application.
+ *
+ * @see [[Yii::beginProfile()]]
+ * @see [[Yii::endProfile()]]
+ *
+ * @author cronfy <cronfy@gmail.com>
+ * @author Qiang Xue <qiang.xue@gmail.com>
+ * @since 2.0.12
+ */
+class Profiler extends Component
+{
+
+    /**
+     * Marks the beginning of a code block for profiling.
+     * This has to be matched with a call to [[end()]] with the same category name.
+     * The begin- and end- calls must also be properly nested.
+     *
+     * @param string $token token for the code block
+     * @param string $category the category of this profiler message (used for logging)
+     * @see end()
+     */
+    public function begin($token, $category = 'application')
```

Comment:
@bashkarev 

1. I agree.
2. I don't agree: Profiler should have enable/disable option instead, not related to debug mode. Consider this use case: you have website that by some reason gives huge load on database **sometimes**. This load does not reproduce in dev environment. Solution is to enable profiler on production and attach a collector to it that will analyze database queries and report statistics about it. Same with CPU load, memory usage.
3. I agree.